### PR TITLE
Format small

### PR DESCRIPTION
**Pull request by uribgp**
Related to #1496

Description: Breaking up giant pull

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by uribgp** [comment]
error: pathspec 'format_small' did not match any file(s) known to git

I'm not sure what that error means.  I probably did something wrong when trying to initiate this pull request.

I am struggling to refactor this: 
```
def log_metrics(logger, epoch, elapsed, tag, metrics):
    logger.info(
        "\nEpoch {} - elapsed: {} - {} metrics:\n {}".format(
            epoch, elapsed, tag, "\n".join(["\t{}: {}".format(k, v) for k, v in metrics.items()])
        )
    )
```

because I don't know what exactly logger.info does.  This is the closest I've come, but I wanted to check before refactoring the others similarly:

```
def log_metrics(logger, epoch, elapsed, tag, metrics):
    logger.info(f"\nEpoch {epoch} - elapsed: {elapsed} - {tag} metrics:"
    "\n".join([f"\t{k}: {v}" for k, v in metrics()]))
```

File: docs/source/faq.rst [hunk 1]
```diff
@@ -375,11 +375,11 @@ Simpliest way to fetch time of single epoch and complete training is to use
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_epoch_time():
-        print("{}: {}".format(trainer.state.epoch, trainer.state.times["EPOCH_COMPLETED"]))
+        print(f"{trainer.state.epoch}: {trainer.state.times["EPOCH_COMPLETED"]}")
```
Comment (vfdev-5) [inline docs/source/faq.rst:378]:
```suggestion
        print(f"{trainer.state.epoch}: {trainer.state.times['EPOCH_COMPLETED']}")
```

File: docs/source/faq.rst [hunk 2]
```diff
@@ -375,11 +375,11 @@ Simpliest way to fetch time of single epoch and complete training is to use
...
 
     @trainer.on(Events.COMPLETED)
     def log_total_time():
-        print("Total: {}".format(trainer.state.times["COMPLETED"]))
+        print(f"Total: {trainer.state.times["COMPLETED"]}")
```
Comment (vfdev-5) [inline docs/source/faq.rst:382]:
```suggestion
        print(f"Total: {trainer.state.times['COMPLETED']}")
```

File: docs/source/quickstart.rst [hunk 3]
```diff
@@ -28,21 +28,19 @@ Code
...
-        print("Epoch[{}] Loss: {:.2f}".format(trainer.state.epoch, trainer.state.output))
+        print(f"Epoch[{trainer.state.epoch}] Loss: {trainer.state.output:.2f}")
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_training_results(trainer):
         evaluator.run(train_loader)
         metrics = evaluator.state.metrics
-        print("Training Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
-              .format(trainer.state.epoch, metrics["accuracy"], metrics["nll"]))
+        print(f"Training Results - Epoch: {trainer.state.epoch}  Avg accuracy: {metrics["accuracy"]:.2f} Avg loss: {metrics["nll"]:.2f}")
```
Comment (vfdev-5) [inline docs/source/quickstart.rst:37]:
```suggestion
        print(f"Training Results - Epoch: {trainer.state.epoch}  Avg accuracy: {metrics['accuracy']:.2f} Avg loss: {metrics['nll']:.2f}")
```

File: docs/source/quickstart.rst [hunk 4]
```diff
@@ -28,21 +28,19 @@ Code
...
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_validation_results(trainer):
         evaluator.run(val_loader)
         metrics = evaluator.state.metrics
-        print("Validation Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
-              .format(trainer.state.epoch, metrics["accuracy"], metrics["nll"]))
+        print("Validation Results - Epoch: {trainer.state.epoch}  Avg accuracy: {metrics["accuracy"]:.2f} Avg loss: {metrics["nll"]:.2f}")
```
Comment (vfdev-5) [inline docs/source/quickstart.rst:43]:
```suggestion
        print("Validation Results - Epoch: {trainer.state.epoch}  Avg accuracy: {metrics['accuracy']:.2f} Avg loss: {metrics['nll']:.2f}")
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @uribgp !
Looks good. A minor fixes is still to apply.

**Comment by vfdev-5** [comment]
There is also a code formatting error : https://github.com/pytorch/ignite/pull/1503/checks?check_run_id=1565551457#step:4:40
Could you please [pass black on the codebase](https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#formatting-without-pre-commit) and commit it.


> because I don't know what exactly logger.info does. 

`logger.info` print given string into std out. Your solution is correct just misses a "\n" to print metrics in another line:
```python
def log_metrics(logger, epoch, elapsed, tag, metrics):
    logger.info(
        f"\nEpoch {epoch} - elapsed: {elapsed} - {tag} metrics: \n" 
        "\n".join([f"\t{k}: {v}" for k, v in metrics()])
    )
```

File: docs/source/quickstart.rst (hunk 4 above)
Comment (uribgp) [reply docs/source/quickstart.rst]: It looks like line 43 is missing the 'f' after the print and before the "Validation

**Comment by uribgp** [comment]
running "black ." on the outer most folder formatted 72 files.  For example in ignite/setup.py line 41:

`packages=find_packages(exclude=("tests", "tests.*",)),`

was changed to `packages=find_packages(
        exclude=(
            "tests",
            "tests.*",
        )
    ),`

Is that supposed to happen?  Looking through your repo, I don't see these changes.  I went ahead and installed pre-commit as well.  Sorry for all the confustion.

**Comment by vfdev-5** [comment]
@uribgp I think you installed a newer version of black. Please install it like `pip install --upgrade "black==19.10b0"`

**Comment by uribgp** [comment]
That was it, thanks a lot!

**Comment by uribgp** [comment]
I wanted to make sure the code snippet we discussed worked before pushing it, so I tested it out like this:
```
def log_metrics_test(logger, epoch, elapsed, tag, metrics):
    print(
        f"\nEpoch {epoch} - elapsed: {elapsed} - {tag} metrics: \n" 
        "\n".join([f"\t{k}: {v}" for k, v in metrics()])
    )
```

The output looked a bit strange:
```       
       1: 1
Epoch epoch string - elapsed: elapsed string - tag string metrics: 

        2: 2
Epoch epoch string - elapsed: elapsed string - tag string metrics: 

        3: 3
Epoch epoch string - elapsed: elapsed string - tag string metrics: 

        4: 4
```

I tried it again with using a variable for the metrics output (no backslashes in f-strings makes it a bit awkward)

```
def log_metrics(logger, epoch, elapsed, tag, metrics):
    metrics_output = "\n".join([f"\t{k}: {v}" for k, v in metrics()])
    print(f"\nEpoch {epoch} - elapsed: {elapsed} - {tag} metrics:\n {metrics_output}")
```

the output was:

```
Epoch epoch string - elapsed: elapsed string - tag string metrics:
        1: 1
        2: 2
        3: 3
        4: 4
```

Which I'm guessing is more like what it should look like.  Just wanted to check before refactoring those.

**Comment by vfdev-5** [comment]
@uribgp thanks for pointing out this question. Yes, your last code snippet is what we would like to have. I also found out that it should be `metrics.items()` (not `metrics()`) in the code. Working code is
```python
epoch = 1
elapsed = 0.123
tag = "Train"
metrics = {
    "Accuracy": 0.89,
    "Loss": 0.12,
}

metrics_output = "\n".join([f"\t{k}: {v}" for k, v in metrics.items()])
print(
    f"\nEpoch {epoch} - elapsed: {elapsed} - {tag} metrics:\n {metrics_output}" 
)

> Epoch 1 - elapsed: 0.123 - Train metrics:
 	Accuracy: 0.89
	Loss: 0.12
```

**Comment by uribgp** [comment]
OK great.  is everything else looking ok for the current files?

**Comment by vfdev-5** [comment]
@uribgp yes it is almost done. Could you please update `log_metrics` according to above discussion here: https://github.com/pytorch/ignite/blob/86d1cc7cb7e85db48236b055bec68ae7949428b2/examples/references/classification/imagenet/code/scripts/training.py#L114-L119 and here: https://github.com/pytorch/ignite/blob/86d1cc7cb7e85db48236b055bec68ae7949428b2/examples/references/segmentation/pascal_voc2012/code/scripts/training.py#L139-L144
Thanks

**Comment by uribgp** [comment]
OK, got it.  Sorry, misunderstood about that; I was going to put it in a later pull.

**Review by vfdev-5 — APPROVED** [review]
Thanks @uribgp !

**Pull request merged**